quicfire_tools/inputs.py

```diff
@@ -722,3 +745,356 @@ def from_file(cls, directory: str | Path):
+        Flag for fuel inputs:
+            - density
+            - moisture
+            - height
+        1 = uniform; 2 = provided thru QF_FuelDensity.inp, 3 = Firetech files
+        for quic grid, 4 = Firetech files for different grid
+        (need interpolation)
+    fuel_params : FuelParams
+        FuelParams class with named fuel parameters for a uniform grid.
+        Defaults to None unless fuel_flag = 1
+    ignition_type: IgnitionType
+        Ignitions shape or source. @anthony
+    ignitions_per_cell: int
+        Number of ignition per cell of the fire model. Recommended max value
+        of 100
+    firebrand_flag : int
+        Firebrand flag, 0 = off; 1 = on. Recommended value = 0 ; firebrands
+        are untested for small scale problems
+    auto_kill : int
+        Kill if the fire is out and there are no more ignitions or firebrands
+        (0 = no, 1 = yes)
+    eng_to_atm_out : int
+        Output flag [0, 1]: gridded energy-to-atmosphere
+        (3D fire grid + extra layers)
+    react_rate_out : int
+        Output flag [0, 1]: compressed array reaction rate (fire grid)
+    fuel_dens_out : int
+        Output flag [0, 1]: compressed array fuel density (fire grid)
+    QF_wind_out : int
+        Output flag [0, 1]: gridded wind (u,v,w,sigma) (3D fire grid)
+    QU_wind_inst_out : int
+        Output flag [0, 1]: gridded QU winds with fire effects, instantaneous
+        (QUIC-URB grid)
+    QU_wind_avg_out : int
+        Output flag [0, 1]: gridded QU winds with fire effects, averaged
+        (QUIC-URB grid)
+    fuel_moist_out : int
+        Output flag [0, 1]: compressed array fuel moisture (fire grid)
+    mass_burnt_out : int
+        Output flag [0, 1]: vertically-integrated % mass burnt (fire grid)
+    firebrand_out : int
+        Output flag [0, 1]: firebrand trajectories. Must be 0 when firebrand
+        flag is 0
+    emissions_out : int
+        Output flag [0, 5]: compressed array emissions (fire grid):
+            0 = do not output any emission related variables
+            1 = output emissions files and simulate CO in QUIC-SMOKE
+            2 = output emissions files and simulate PM2.5 in QUIC- SMOKE
+            3 = output emissions files and simulate both CO and PM2.5 in
+                QUIC-SMOKE
+            4 = output emissions files but use library approach in QUIC-SMOKE
+            5 = output emissions files and simulate both water in QUIC-SMOKE
+    radiation_out : int
+        Output flag [0, 1]: gridded thermal radiation (fire grid)
+    intensity_out : int
+        Output flag [0, 1]: surface fire intensity at every fire time step
+    """
+    filename: str = Field("QUIC_fire.inp", allow_mutation=False)
+    nz: PositiveInt
+    time_now: PositiveInt
+    sim_time: PositiveInt
+    output_times: PositiveInt | OutputTimes
+    ignition_type: IgnitionType = IgnitionType(ignition_flag=6)
+    fire_flag: Literal[0, 1] = 1
+    random_seed: PositiveInt = random.randrange(1000)
```

Comment:
Can we just default this to -1? Or, 42 as I usually do for my random seed. I'd rather have a default value than a random number here.

Reply:
So the problem I was running into was with pydantic. I could use `Field(ge=1, default=-1)` which would set the value to -1 when instantiating the class, but after that the value could not be set to -1, ie when using from_file. So I just set a random number since that's what -1 does anyway. But an alternative to allow -1 as an input would be to do `Field(ge=-1)` and then add a little field_validator that doesn't allow zero. Should we do that?